src/code/z_schedule.c

```diff
@@ -0,0 +1,282 @@
+#include "global.h"
+
+#define SCHEDULE_CALC_TIME(hour, minute, dest, temp) \
+    (temp) = (hour)*60.0f;                           \
+    (temp) += (minute);                              \
+    (dest) = (temp) * (0x10000 / 60 / 24.0f);        \
+    (dest) = SCHEDULE_CONVERT_TIME(dest);
+
+s32 Schedule_CheckFlagS(GlobalContext* globalCtx, u8** script, ScheduleResult* result) {
+    ScheduleCmdCheckFlagS* cmd = (ScheduleCmdCheckFlagS*)*script;
+    u16 flag = (cmd->flagByte << 8) | cmd->flagMask;
+
+    if (gSaveContext.weekEventReg[flag >> 8] & (flag & 0xFF)) {
+        *script = *script + cmd->offset;
```

Comment:
```suggestion
        *script += cmd->offset;
```
Maybe it's different with dereferencing pointers, but assuming it matches, that can be applied to a lot of cases in this file